### DISCUSSION TITLE
custom sasl login callback handler

### DISCUSSION THREAD

**Discussion by MaayanFarchi** [strimzi/strimzi-kafka-operator#5820] [Q&A]
Hey there, 

We are using Strimzi MM2 for event hub. As for today we would like to change the authentication to use MSI (Managed Identity) 
Generally speaking this can be done writing a custom sasl login call back handler, the question is will we be able to integrate that with Strimzi?

Thanks in advance!

**Comment by scholzj** [comment]
I have no idea what MSI is. But we currently don't support configuring custom authentication. If needed, you can open an enhancement issue or a proposal for this. There is also a blog post describing how you can connect MM2 to Azure Event Hub: https://strimzi.io/blog/2020/06/09/mirror-maker-2-eventhub/

**Reply by MaayanFarchi** [reply]
Our issue is not with connecting MM2 to azure EH, we would just like to authenticate in a different way. I will see if I have a proposal of how to support custom authentication, thanks @scholzj.

**Reply by scholzj** [reply]
There is currently a similar proposal for server side authentication: https://github.com/strimzi/proposals/pull/41 ... so feel free to have a look at it to see what it might include. I would assume that the client side might be a bit easier to implement than the server side. There were already similar requirement for Amazon MSK authentication support (https://github.com/strimzi/strimzi-kafka-operator/issues/5473). So it seems we have more similar requirements which it should cover.
